src/main/java/seedu/address/logic/commands/FilterCommand.java

```diff
@@ -27,8 +31,12 @@ public FilterCommand(StudentHasLabPredicate predicate) {
     }
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+        Lab toFilter = predicate.getLab();
+        if (!model.hasLab(toFilter)) {
+            throw new CommandException(String.format(MESSAGE_LAB_NOT_FOUND, toFilter.labNumber));
+        }
```

Comment:
Just to confirm that you're trying to check if a lab is in the system based on lab number only. The `hasLab` function uses `isSameLab` which only checks the lab number. Just checking if this was intentional or you had something else in mind 🙃

Reply:
Yeap, i think only need to check lab number here.